lib/wda/utils.js

```diff
@@ -156,6 +157,47 @@ CODE_SIGN_IDENTITY = ${signingId}
   return xcconfigPath;
 }
 
+/**
+ * Creates xctestrun file per device & platform version.
+ * We expects to have WebDriverAgentRunner_iphoneos${platformVersion}-arm64.xctestrun for real device
+ * and WebDriverAgentRunner_iphonesimulator${platformVersion}-x86_64.xctestrun for simulator located @bootstrapPath
+ *
+ * @param {boolean} isRealDevice - Equals to true if the current device is a real device
+ * @param {string} udid - The device UDID.
+ * @param {string} platformVersion - The platform version of OS.
+ * @param {string} bootstrapPath - The device UDID.
+ * @param {string} wdaRemotePort - The remote port WDA is listening on.
+ */
+async function setXCTESTRUNFile (isRealDevice, udid, platformVersion, bootstrapPath, wdaRemotePort) {
+  let xctestrunDeviceFileName = `${udid}_${platformVersion}.xctestrun`;
+  let xctestrunFilePath = path.resolve(bootstrapPath, xctestrunDeviceFileName);
+
+  if (!await fs.exists(xctestrunFilePath)) {
+    let xctestBaseFileName = isRealDevice ? `WebDriverAgentRunner_iphoneos${platformVersion}-arm64.xctestrun` :
+      `WebDriverAgentRunner_iphonesimulator${platformVersion}-x86_64.xctestrun`;
+    let originalXctestrunFile = path.resolve(bootstrapPath, xctestBaseFileName);
+    if (!await fs.exists(originalXctestrunFile)) {
+      log.errorAndThrow(`if you are using useXCTESTRUNFile capability then you need to have ${originalXctestrunFile} file`);
+    }
+    await fs.copyFile(originalXctestrunFile, xctestrunFilePath);
```

Comment:
please add an explanation comment here

Reply:
I did put in method signature explanation. I added more detailed explaination here as well.